subprojects/core/src/main/java/org/gradle/configuration/DeferredProjectEvaluationCondition.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configuration;
+
+import org.gradle.api.NonNullApi;
+import org.gradle.api.internal.GradleInternal;
+import org.gradle.internal.buildtree.BuildModelParameters;
+
+@NonNullApi
+public class DeferredProjectEvaluationCondition {
```

Comment:
ℹ️  Shared condition to ensure that project-evaluated listeners are always honored. This is a refactoring to avoid problems in the future, as the condition is used in more than one place.